### PR TITLE
fix(contrib/azure)-replace discovery url

### LINKED ISSUE #4679: Bug (contrib/azure) - unable to add new discovery URL to azure-user-data
The command `./create-azure-user-data $(curl -s https://discovery.etcd.io/new)` did not correctly parse the `user-data-template` file to replace value of DISCOVERY_URL in the `azure-user-data` file. The end result was  `discovery: null`. This was also mentioned in #4593

### PR DESCRIPTION
**Pull request by ritazh**
closes #4679 

Fix bug related to a null value discovery URL that was blocking deployment of Deis to Azure

### CODE REVIEW

**Comment by deis-admin** [comment]
Thanks for the contribution! Please ensure your commits follow our [style guide](http://docs.deis.io/en/latest/contributing/standards/#commit-style-guide). This code will be tested once a Deis maintainer reviews it.

Your Pull Request can earn you credit at DigitalOcean! See http://docs.deis.io/en/latest/contributing/hacking/#digitalocean-credit for details.

File: contrib/coreos/user-data.example [hunk 1]
```diff
@@ -3,7 +3,7 @@
 coreos:
   etcd2:
     # generate a new token for each unique cluster from https://discovery.etcd.io/new
-    discovery: #DISCOVERY_URL
+    discovery: $DISCOVERY_URL
```
Comment (mboersma) [inline contrib/coreos/user-data.example]: This change will break [`make discovery-url`](https://github.com/deis/deis/blob/master/Makefile#L37), which is how  other providers get an `etcd` discovery URL.

**Comment by mboersma** [comment]
I don't see how this would fix anything. `replace()` is just a python string function, so changing what it searches for from `#DISCOVERY_URL` to `$DISCOVERY_URL` is a no-op. `$DISCOVERY_URL` isn't being interpreted as a shell variable in the python code, nor in user-data, so the issue must be something different. (And this change breaks other providers as noted above.)

Could you enter a GitHub issue with details of the actual Azure failure? Maybe that would help pin it down and fix it.

**Comment by sgoings** [comment]
Here's an alternative implementation: https://github.com/sgoings/deis/blob/rigger-azure/contrib/azure/create-azure-user-data#L55-L66

The problem is that the comment in YAML (#DISCOVERY_URL) is stripped out (discovery's value is empty) by yaml.safe_load on newer versions of PyYAML.

**Comment by sgoings** [comment]
@mboersma she did (#4679)

**Comment by mboersma** [comment]
> the comment in YAML (#DISCOVERY_URL) is stripped out

Ok, then this change would help I assume, but it needs to be made and tested against all providers.

**Comment by ritazh** [comment]
@mboersma @sgoings Thanks for the feedback. What is the best way to test this against all the providers?

**Comment by sgoings** [comment]
Looks bueno @ritazh. Just need a rebase + squash to 1 commit!

**Comment by ritazh** [comment]
@sgoings Done!

**Comment by mboersma** [comment]
It appears Deis needs a similar fix for GCE to #4588.

**Pull request merged**